docs/src/development/proposals/MEP9/README.md

```diff
@@ -0,0 +1,66 @@
+# No Open Ports To the Data Center
+
+Our metal-stack partitions typically have open ports for metal-stack native services, these are:
+
+- SSH port on the firewalls
+- bmc-reverse-proxy for serial console access through the metal-console
+
+These open ports are potential security risks. For example, while SSH access is possible only with private key it's still vulnerable to DoS attack.
+
+Therefore, we want to get rid off these open ports to reduce the attack surface to the data center.
+
+## Requirements
+
+- Access to firewall SSH only via VPN
+- Easy to update VPN components
+
+As a next step, we can also consider joining the management servers to the VPN mesh, which would replace typical WireGuard setups for operators to enter resources inside the partition.
+
+## High Level Design
+
+[](./architecture.drawio.svg)
+
+> Simplified drawing showing old vs. new architecture.
+
+TODO:
+
+Store current Headscale and desired Tailscale(`metalctl`?) version in some K8s resource? And allow to update it via `metalctl`.
+
+### Concerns
+
+There's few concerns when using WireGuard for implementing VPN:
+
+1. WireGuard doesn't implement dynamic cipher substitution. Which is important in case one of the crypto methods, used by WireGuard will be broken. The only possible solution for that will be to update WireGuard to a fixed version.
+2. Coordination server(Headscale) is a single point of failure. In case it fails, it potentially can disconnect existing members of the network, as WireGuard can't manage dynamic IPs by itself.
+3. Headscale is already falls behind Tailscale coordination server implementation. Which can complicate the upgrade to newer version of Tailscale client in case of emergency.
+
+### Solutions to concerns
+
+1. Tailscale node software is using userspace implementation of WireGuard -- `wireguard-go`. One of the options is to inject Tailscale client into `metalctl`. And make it available as `metalctl vpn` or similar command. It should be possible to do as `tailscale` node is already available as open sourced Go pkg. That would allow us to control, what version of Tailscale users are using and in case of any critical changes to enforce them to update `metalctl` to use VPN functionality.
+2. Would it be a considerable risk? We could look into `wg-dynamic` project to cover this problem.
+
```

Comment:
Regarding 3.: The repository seems to be well maintained and due to it's open source nature it would also be possible to contribute fixes if it's important to us.